drivers/backup/portworx/portworx.go

```diff
@@ -1535,6 +1592,97 @@ func (p *portworx) DeleteBackupSchedulePolicy(orgID string, policyList []string)
 	return nil
 }
 
+// GetTokenClaimsFromCtx returns JWT claims from the outgoing metadata of the given context
+func GetTokenClaimsFromCtx(ctx context.Context) (jwt.MapClaims, error) {
+	md, ok := metadata.FromOutgoingContext(ctx)
+	if !ok {
+		return nil, fmt.Errorf("no metadata found in context")
+	}
+	authHeaders, ok := md[backup.AuthHeader]
+	if !ok || len(authHeaders) == 0 {
+		return nil, fmt.Errorf("no authorization header found")
+	}
+	tokenString := strings.TrimPrefix(authHeaders[0], "bearer ")
+	token, _, err := new(jwt.Parser).ParseUnverified(tokenString, jwt.MapClaims{})
+	if err != nil {
+		return nil, err
+	}
+	claims, ok := token.Claims.(jwt.MapClaims)
+	if !ok {
+		return nil, fmt.Errorf("invalid JWT claims")
+	}
+	return claims, nil
+}
+
+// GetGivenNameFromCtx extracts and decodes the JWT token from the outgoing context and then returns the given name
+func GetGivenNameFromCtx(ctx context.Context) (string, error) {
+	claims, err := GetTokenClaimsFromCtx(ctx)
+	if err != nil {
+		return "", err
+	}
+	givenName, ok := claims["given_name"].(string)
```

Comment:
Why can't we use `preferred_username` instead of `given_name` ?

Reply:
I feel that preferred_name is optional. I am not sure what the difference is, though.